python/fast_mlsirm/interaction_map.py

```
@@ -0,0 +1,88 @@
+) -> ResidualInteractionMap:
+    """Factor ``observed - expected`` using Gabriel symmetric scaling.
+
+    Missing observed cells are represented by ``NaN`` and excluded through a
+    complete-case rectangle; they are never filled with zero. ``axis_count`` is
+    required because the consuming measurement contract, not this library,
+    determines how many reader-visible axes are retained.
+
+    References:
+        Gabriel, K. R. (1971). The biplot graphic display of matrices with
+            application to principal component analysis. *Biometrika, 58*(3),
+            453–467. https://doi.org/10.1093/biomet/58.3.453
```

Ruff (0.16.2)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)